### PR TITLE
MAINT: refactoring user, chat -> user_id, chat_id

### LINKED ISSUE #174: Rename chat and user arguments to chat_id and user_id
After discussion in [aiogram_ru#80199](https://t.me/aiogram_ru/80199) and #172 is planned to rename all `chat` and `user` arguments to `chat_id` and `user_id` and marking old arguments as deprecated until 3.0.

Affected:

1. https://github.com/aiogram/aiogram/blob/c6923e331c1222b275aa87ce87ffabf17b185890/aiogram/dispatcher/dispatcher.py#L860-L862
2. https://github.com/aiogram/aiogram/blob/c6923e331c1222b275aa87ce87ffabf17b185890/aiogram/dispatcher/dispatcher.py#L887
3. https://github.com/aiogram/aiogram/blob/c6923e331c1222b275aa87ce87ffabf17b185890/aiogram/dispatcher/dispatcher.py#L964

### PR DESCRIPTION
**Pull request by tohabyuraev**
# Description

This pull request rename `user` and `chat` arguments to `user_id` and `chat_id` in many files of **aiogram**.

Fixes #174 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# How Has This Been Tested?

**Test Configuration**:
* Operating System: win32
* Python version: 3.8.2
* pytest-5.4.3, py-1.9.0, pluggy-0.13.1

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by tohabyuraev** [comment]
@Birdi7 I carefully checked the directories:
- `aiogram.contrib`
- `aiogram.dispatcher`
- `aiogram.utils`
- `examples`

and some modules from `aiogram.bot`:
- `__init__.py`
- `api.py`
- `base.py`.

In `bot.py` I did not found code with `user` and `chat`, but I am not sure. Next time I can check this.
In `aiogram.types`, I already checked about half of all the modules.

So, the next step is to rewrite tests. But there are some changes like remove `aiomongo`. I need to know what should I do.

File: aiogram/contrib/fsm_storage/memory.py [hunk 1]
```diff
@@ -20,8 +20,8 @@ async def close(self):
     def __init__(self):
         self.data = {}
 
-    def resolve_address(self, chat, user):
-        chat_id, user_id = map(str, self.check_address(chat=chat, user=user))
+    def resolve_address(self, chat_id, user_id):
```
Comment (Birdi7) [inline aiogram/contrib/fsm_storage/memory.py:23]: Here and later: add 2 decorators
```python3
    @renamed_argument(old_name='user', new_name='user_id', until_version='3.0', stacklevel=3)
    @renamed_argument(old_name='chat', new_name='chat_id', until_version='3.0', stacklevel=4)
```
to all functions, in which you change the parameter names. This is required to keep the backward compatibility. We plan to completely remove this naming only in 3.0 (or not later than 2.12), so we need to allow both `user` and `user_id` for now

File: aiogram/contrib/fsm_storage/mongo_aiomongo.py [hunk 2]
```diff
@@ -92,7 +92,7 @@ async def get_db(self) -> Database:
     @staticmethod
     async def apply_index(db):
         for collection in COLLECTIONS:
-            await db[collection].create_index(keys=[('chat', 1), ('user', 1)],
+            await db[collection].create_index(keys=[('chat_id', 1), ('user_id', 1)],
```
Comment (Birdi7) [inline aiogram/contrib/fsm_storage/mongo_aiomongo.py:95]: This file will be removed, so I'll leave comments only in `mongo_motor.py`. Feel free to delete this file, and move content of `mongo_motor.py` to `mongo.py`

File: aiogram/contrib/fsm_storage/mongo_motor.py [hunk 3]
```diff
@@ -104,7 +104,7 @@ async def get_db(self) -> AsyncIOMotorDatabase:
     @staticmethod
     async def apply_index(db):
         for collection in COLLECTIONS:
-            await db[collection].create_index(keys=[('chat', 1), ('user', 1)],
+            await db[collection].create_index(keys=[('chat_id', 1), ('user_id', 1)],
```
Comment (Birdi7) [inline aiogram/contrib/fsm_storage/mongo_motor.py:107]: hmm. such renaming of internal database fields can cause a huge problems to users of MongoDB. Let's keep the internal naming as is

File: aiogram/contrib/fsm_storage/mongo_motor.py [hunk 4]
```diff
@@ -114,75 +114,90 @@ async def close(self):
...
+                        chat_id: Union[str, int, None] = None,
+                        user_id: Union[str, int, None] = None,
                         state: Optional[AnyStr] = None):
-        chat, user = self.check_address(chat=chat, user=user)
+        chat_id, user_id = self.check_address(chat_id=chat_id, user_id=user_id)
         db = await self.get_db()
 
         if state is None:
-            await db[STATE].delete_one(filter={'chat': chat, 'user': user})
+            await db[STATE].delete_one(filter={'chat_id': chat_id, 'user_id': user_id})
```
Comment (Birdi7) [inline aiogram/contrib/fsm_storage/mongo_motor.py:125]: same about internal naming

File: aiogram/contrib/fsm_storage/mongo_motor.py [hunk 5]
```diff
@@ -114,75 +114,90 @@ async def close(self):
...
         else:
-            await db[STATE].update_one(filter={'chat': chat, 'user': user},
+            await db[STATE].update_one(filter={'chat_id': chat_id, 'user_id': user_id},
```
Comment (Birdi7) [inline aiogram/contrib/fsm_storage/mongo_motor.py:127]: same about internal naming

File: aiogram/contrib/fsm_storage/mongo_motor.py [hunk 6]
```diff
@@ -114,75 +114,90 @@ async def close(self):
...
-    async def get_state(self, *, chat: Union[str, int, None] = None, user: Union[str, int, None] = None,
+    async def get_state(self, *,
+                        chat_id: Union[str, int, None] = None,
+                        user_id: Union[str, int, None] = None,
                         default: Optional[str] = None) -> Optional[str]:
-        chat, user = self.check_address(chat=chat, user=user)
+        chat_id, user_id = self.check_address(chat_id=chat_id, user_id=user_id)
         db = await self.get_db()
-        result = await db[STATE].find_one(filter={'chat': chat, 'user': user})
+        result = await db[STATE].find_one(filter={'chat_id': chat_id, 'user_id': user_id})
```
Comment (Birdi7) [inline aiogram/contrib/fsm_storage/mongo_motor.py:136]: same about internal naming

File: aiogram/contrib/fsm_storage/mongo_motor.py [hunk 7]
```diff
@@ -114,75 +114,90 @@ async def close(self):
...
+    async def set_data(self, *,
+                       chat_id: Union[str, int, None] = None,
+                       user_id: Union[str, int, None] = None,
                        data: Dict = None):
-        chat, user = self.check_address(chat=chat, user=user)
+        chat_id, user_id = self.check_address(chat_id=chat_id, user_id=user_id)
         db = await self.get_db()
 
-        await db[DATA].update_one(filter={'chat': chat, 'user': user},
+        await db[DATA].update_one(filter={'chat_id': chat_id, 'user_id': user_id},
```
Comment (Birdi7) [inline aiogram/contrib/fsm_storage/mongo_motor.py:147]: same about internal naming, and later in this file

**Review by Birdi7 — DISMISSED** [review]
The done work is impressive. However, some fixes should be made in order that we will be able to merge it. 

Talking about upcoming changes to the project, the aiomongo removal will be merged as soon as @JrooTJunior will take a look on it. Just merge the later `dev-2.x` to be sure you're up.

File: .gitignore [hunk 8]
```diff
@@ -52,6 +52,15 @@ ENV/
 # JetBrains
 .idea/
 
+# VS Code
+.vscode
+__pycache__
```
Comment (JrooTJunior) [inline .gitignore:57]: `__pycache__` is already covered by line 3 in this file. This line is unnecessary.

File: aiogram/contrib/fsm_storage/memory.py (hunk 1 above)
Comment (JrooTJunior) [reply aiogram/contrib/fsm_storage/memory.py:23]: Yep. Changing the naming of arguments in framework is breaking change and can not be done without deprecation process in minor releases.

**Review by JrooTJunior — DISMISSED** [review]

**Comment by Birdi7** [comment]
I'll remove label `breaking`, as this PR is meant to be a non-breaking refactoring for version 2 of the framework

**Comment by tohabyuraev** [comment]
@Birdi7 Question for `aiogram.utils.exceptions.Throttled`. It seems that there will not be a mistake without decorator.

**Comment by Birdi7** [comment]
> @Birdi7 Question for `aiogram.utils.exceptions.Throttled`. It seems that there will not be a mistake without decorator.

Just leave it as it

**Comment by tohabyuraev** [comment]
And what should I do with `FSMContext` and `DisabledStorage`? Is it nessesary to add decorators?

**Comment by tohabyuraev** [comment]
@Birdi7 @JrooTJunior Are there any comments?

**Comment by Birdi7** [comment]
For me the PR looks ok, Alex will check on his own

**Comment by JrooTJunior** [comment]
Lgtm, but i need to check it manually. I will do it soon as possible

**Comment by JrooTJunior** [comment]
v2 will no more be supported.

**Pull request closed without merging**